packages/api/cms-api/src/dam/files/folders.service.ts

```diff
@@ -217,6 +219,42 @@ export class FoldersService {
         return result === 1;
     }
 
+    async getFolderPosition(folderId: string, args: DamFolderListPositionArgs): Promise<number> {
+        const subQb = withFoldersSelect(
+            this.foldersRepository
+                .createQueryBuilder("folder")
+                .select(`folder.id, ROW_NUMBER() OVER( ORDER BY folder."${args.sortColumnName}" ${args.sortDirection} ) AS row_number`),
+            {
+                includeArchived: args.includeArchived,
+                parentId: args.parentId,
+                query: args.filter?.searchText,
+                sortColumnName: args.sortColumnName,
+                sortDirection: args.sortDirection,
+            },
+        );
+
+        const connection = this.orm.em.getConnection();
+        const metadata = this.orm.em.getMetadata();
+
+        const folderTableName = metadata.get(Folder.name).tableName;
+
+        const result: Array<{ row_number: string }> = await connection.execute(
```

Comment:
Because I have no idea how to join on a subquery using the mikroorm query builder. 
I tried around but couldn't find a way. In the mikroorm docs, the only example is a simple subquery in a where clause (https://mikro-orm.io/docs/query-builder#using-sub-queries). The only thing I found on Google is one Stackoverflow question without an answer (https://stackoverflow.com/questions/69622374/how-can-i-perform-a-join-with-a-subquery-using-mikro-orm)